test/support/test_config.h

```diff
@@ -293,11 +293,14 @@
 #    error "TEST_ONLY_HETERO_POLICIES is passed but device backend is not available"
 #endif
 
-// There are issues with stable_sort for libcpp with iterators with a deleted comma operator, disable those tests
-#if (defined(_LIBCPP_VERSION))
-#   define _PSTL_LIBCPP_NO_COMMA_TESTS_BROKEN 1
+//There are issues with stable_sort for libcpp with iterators with a deleted comma operator, disable no_comma compile
+// only tests. Also, since this adds significant amount of code compilation to the build, lets be more conservative
+// about when we try to test this. Since debug build mode and unnamed lambda support dont change the code in any way
+// which should interact with what we are testing for here, lets disable it for those cases for time / build space.
+#if !defined(_LIBCPP_VERSION) && !defined(_DEBUG) && defined(__SYCL_UNNAMED_LAMBDA__)

```

Comment:
With regards to detecting debug configs, I know `_DEBUG` is defined by MSVC. Do other compilers define it as well? In our CMakeLists.txt, we define `PSTL_USE_DEBUG=1` for CMake debug configurations, so perhaps this is a more reliable macro to check.